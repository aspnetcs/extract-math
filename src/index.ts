export interface Segment {
  type: "text" | "display" | "inline"
  math: boolean
  value: string
}

export function extractMath(text: string): Segment[] {
  return [
    {
      type: "text",
      math: false,
      value: text
    }
  ]
}
