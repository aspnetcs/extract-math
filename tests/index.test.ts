import { extractMath } from "../src"

test("empty input", () => {
  const segments = extractMath("")
  expect(segments).toEqual([
    {
      type: "text",
      math: false,
      value: ""
    }
  ])
})

test("plain text", () => {
  const segments = extractMath("hello world")
  expect(segments).toEqual([
    {
      type: "text",
      math: false,
      value: "hello world"
    }
  ])
})
