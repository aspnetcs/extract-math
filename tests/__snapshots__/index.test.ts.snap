// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`combined text 1`] = `
Array [
  Object {
    "math": false,
    "type": "text",
    "value": "
    Text with an inline formula ",
  },
  Object {
    "math": true,
    "type": "inline",
    "value": "x ^ 2 + 5",
  },
  Object {
    "math": false,
    "type": "text",
    "value": " and a displayed equation:

    ",
  },
  Object {
    "math": true,
    "type": "display",
    "value": "\\\\sum_{i=1}^n(x_i^2 - \\\\overline{x}^2)",
  },
  Object {
    "math": false,
    "type": "text",
    "value": "

    The inline formula is represented as $expression$ and the displayed equation as $$expression$$.
    The $ symbol can be escaped like this: \\\\$
  ",
  },
]
`;
